### ISSUE TITLE
SharpYaml serialization can't handle System.Guid

### ISSUE THREAD

**Issue by PathogenDavid** [xoofx/SharpYaml#26]
When I was evaluating SharpYaml for my needs, I found that it doesn't handle GUIDs. Rather than serialize the GUID, it serializes and empty object. I would at least expect it to throw an exception, alerting me to the fact that the serialized data is incomplete.

Below is a minimal program to demonstrate the issue. The issue occurs both when the GUID is a member of a type being serialized and when it is serialized alone.

```C#
using System;
using SharpYaml.Serialization;

namespace YamlPlayground
{
    class SharpYamlGuidTest
    {
        public struct MyStructWithGuid
        {
            public Guid MyGuid;

            public MyStructWithGuid(Guid myGuid)
                => MyGuid = myGuid;
        }

        public static void SerializeGuidWithSharpYaml()
        {
            var serializerSettings = new SerializerSettings()
            {
                EmitTags = false
            };
            var serializer = new Serializer(serializerSettings);

            Guid guid = Guid.Parse("f8dd32bf-ebb3-4a64-982e-9450398cfaea");

            string yaml1 = serializer.Serialize(guid);
            Console.WriteLine(yaml1);
            // Actual: 
            //  {}
            // Expected:
            //  f8dd32bf-ebb3-4a64-982e-9450398cfaea
            //  ...

            string yaml2 = serializer.Serialize(new MyStructWithGuid(guid));
            Console.WriteLine(yaml2);
            // Actual: 
            //  MyGuid: {}
            // Expected:
            //  MyGuid: f8dd32bf-ebb3-4a64-982e-9450398cfaea

            Guid guid1 = serializer.Deserialize<Guid>(yaml1);
            Guid guid2 = serializer.Deserialize<MyStructWithGuid>(yaml2).MyGuid;

            Console.WriteLine(guid1 == guid);
            Console.WriteLine(guid2 == guid);
            // Actual:
            //  false
            //  false
            // Expected:
            //  true
            //  true
        }
    }
}
```

**Comment by PathogenDavid** [comment]
Not surprisingly, making a custom serializer for this was pretty trivial.

Also, after thinking about it more, the serializer clearly doesn't choke on GUID since it has no public properties/fields. I suppose the only way it would've thrown an exception was if it threw when a type has private fields and no public fields. Although you could argue it should complain if there's private fields at all since they will never be serialized.

@xoofx If you don't think System.Guid is out of scope of internally supported serializers, I could package the below up as a pull request. Otherwise, this issue can just live on for people to find when they search for GUID issues.

```C#
using System;
using SharpYaml;
using SharpYaml.Events;
using SharpYaml.Serialization;
using SharpYaml.Serialization.Serializers;

namespace YamlPlayground
{
    class GuidSerializer : ScalarSerializerBase, IYamlSerializable
    {
        public override object ConvertFrom(ref ObjectContext context, Scalar fromScalar)
            => Guid.Parse(fromScalar.Value);

        public override string ConvertTo(ref ObjectContext objectContext)
        {
            object value = objectContext.Instance;

            if (value is Guid guid)
            { return guid.ToString(); }

            throw new YamlException($"Unable to serialize scalar [{value}] not supported");
        }
    }
}
```

**Comment by xoofx** [comment]
Yep, Guid should probably part of the default types, so PR welcome

**Comment by xoofx** [comment]
Fixed by commit 1230ce2
